buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/PerformanceTest.java

```diff
@@ -79,44 +132,82 @@ public String getRuns() {
     @Option(option = "checks", description = "Tells which regressions to check. One of [none, speed, all]")
     public void setChecks(@Nullable String checks) {
         this.checks = checks;
-        systemProperty("org.gradle.performance.execution.checks", checks);
     }
 
     @Nullable @Optional @Input
     public String getChecks() {
         return checks;
     }
 
-    @Nullable @Optional @Input
-    public String getChannel() {
-        return channel;
-    }
-
     @Option(option = "channel", description = "Channel to use when running the performance test. By default, 'commits'.")
     public void setChannel(@Nullable String channel) {
         this.channel = channel;
-        systemProperty("org.gradle.performance.execution.channel", channel);
     }
 
-    @OutputDirectory
-    public File getDebugArtifactsDirectory() {
-        return debugArtifactsDirectory;
+    @Nullable @Optional @Input
+    public String getChannel() {
+        return channel;
     }
 
-    public void setDebugArtifactsDirectory(File debugArtifactsDirectory) {
-        this.debugArtifactsDirectory = debugArtifactsDirectory;
+    public void setResultStoreClass(String resultStoreClass) {
+        this.resultStoreClass = resultStoreClass;
+    }
+
+    @Nullable @Optional @Input
+    public String getResultStoreClass() {
+        return resultStoreClass;
     }
 
     @Option(option = "flamegraphs", description = "If set to 'true', activates flamegraphs and stores them into the 'flames' directory name under the debug artifacts directory.")
     public void setFlamegraphs(String flamegraphs) {
-        if ("true".equals(flamegraphs)) {
+        this.flamegraphs = "true".equals(flamegraphs);
```

Comment:
We could use `Boolean.parseBoolean` here, right?

Reply:
Right.